src/httpserver.cpp

```diff
@@ -667,12 +667,21 @@ std::optional<std::string> HTTPRequest::GetQueryParameter(const std::string& key
 {
     const char* uri{evhttp_request_get_uri(req)};
 
-    return GetQueryParameterFromUri(uri, key);
+    try {
+        return GetQueryParameterFromUri(uri, key);
+    } catch (const std::runtime_error& e) {
+        LogPrint(BCLog::HTTP, "%s\n", e.what());
+        // add additional information and rethrow the exception as we can't send a reply from the worker
+        throw std::runtime_error(strprintf("Error occurred trying to get query param key \"%s\": ", key) + e.what());
+    }
```

Comment:
What's the benefit of re-handling this error? The error message from `GetQueryParameterFromUri` is enough (or could be updated there to reference query param fetching, but it doesn't really have anything to do with the URI being invalid, so I'd prefer leaving it out). 

<details>
<summary>git diff</summary>

```diff
diff --git a/src/httpserver.cpp b/src/httpserver.cpp
index eb1f0bd93..3f0d82497 100644
--- a/src/httpserver.cpp
+++ b/src/httpserver.cpp
@@ -667,13 +667,7 @@ std::optional<std::string> HTTPRequest::GetQueryParameter(const std::string& key
 {
     const char* uri{evhttp_request_get_uri(req)};
 
-    try {
-        return GetQueryParameterFromUri(uri, key);
-    } catch (const std::runtime_error& e) {
-        LogPrint(BCLog::HTTP, "%s\n", e.what());
-        // add additional information and rethrow the exception as we can't send a reply from the worker
-        throw std::runtime_error(strprintf("Error occurred trying to get query param key \"%s\": ", key) + e.what());
-    }
+    return GetQueryParameterFromUri(uri, key);
 }
 
 std::optional<std::string> GetQueryParameterFromUri(const char* uri, const std::string& key)
diff --git a/test/functional/interface_rest.py b/test/functional/interface_rest.py
index fa6193039..fb647a62f 100755
--- a/test/functional/interface_rest.py
+++ b/test/functional/interface_rest.py
@@ -282,7 +282,7 @@ class RESTTest (BitcoinTestFramework):
 
         # Check invalid uri (% symbol at the end of the request)
         resp = self.test_rest_request(f"/headers/{bb_hash}%", ret_type=RetType.OBJ, status=400)
-        assert_equal(resp.read().decode('utf-8').rstrip(), f"Error occurred trying to get query param key \"count\": Invalid uri detected while parsing \"/rest/headers/{bb_hash}%.json\"")
+        assert_equal(resp.read().decode('utf-8').rstrip(), f"Invalid uri detected while parsing \"/rest/headers/{bb_hash}%.json\"")
 
         # Compare with normal RPC block response
         rpc_block_json = self.nodes[0].getblock(bb_hash)

```
</details>

Reply:
Yeah, the intention was to add the param key on top of the previous error, adding more details at this level but that info is also available from the caller of `GetQueryParameterFromUri`. Ok, I'll leave it out.